### ISSUE TITLE
Create bitbucket-sdm

### ISSUE THREAD

**Issue by johnsonr** [atomist/github-sdm#225]
Split out new `bitbucket-sdm` project, based on extracted `sdm-core`

This will include

- BitBucket project operation support
- Full delivery support

JIRA support out of scope for now

**Comment by jessitron** [comment]
is it possible to do it the other way around? ... a VcsProvider interface that implements what the SDM needs?

**Comment by jwalter** [comment]
Is it correct to assume this only includes BitBucket cloud, not self-hosted?

**Comment by cdupuis** [comment]
No, this issues is actually covering both, BitBucket Cloud and self hosted.
